src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -307,28 +308,45 @@ class SlickDataAccess(databaseConfig: Config, val dataAccess: DataAccessComponen
     runTransaction(action)
   }
 
-  override def getWorkflowsByState(states: Traversable[WorkflowState]): Future[Traversable[WorkflowInfo]] = {
+  override def getWorkflow(workflowId: WorkflowId): Future[WorkflowDescriptor] = {
+    val action = for {
+      workflowExecutionResult <- dataAccess.workflowExecutionsByWorkflowExecutionUuid(workflowId.toString).result.head
```

Comment:
I assume boom.  I recall this was a design choice in `SlickDataAccess`, this pattern is everywhere.